### DISCUSSION TITLE
Problem when using the new backgroundShader and AxisValueOverrider.adaptiveYValues together

### DISCUSSION THREAD

**Discussion by jibe-n** [patrykandpatrick/vico#487] [Questions] [answered]
Hi!

I'm currently upgrading Vico from v1.13 to v2.0.0-alpha.2 in my project and I'm encountering a problem for which I can't find a solution.

Previously, I had an `axisValuesOverrider` set to `AxisValueOverrider.adaptiveYValues(yFraction = 1f)` to get the scale I needed for my chart (as suggested in [this](https://github.com/patrykandpatrick/vico/issues/171) issue) coupled with a `lineBackgroundShader = verticalGradient(colors = arrayOf(...colors))`.

With this combination, I was getting the result I expect:

![Screenshot_20231211_134640_Crypto Market Cap](https://github.com/patrykandpatrick/vico/assets/82245793/8b71deed-3fb1-45a3-be9a-905c5fc1de8d)

However, after upgrading to Vico v2.0.0-alpha.2, I changed my `lineBackgroundShader` to `backgroundShader = DynamicShaders.fromBrush(Brush.verticalGradient(listOf(...colors)))` and I'm now getting this result:

![Screenshot_20231211_134548_Crypto Market Cap](https://github.com/patrykandpatrick/vico/assets/82245793/41ee3f5c-c8b6-4f8e-9921-99cfef0a7618)

I understand the `backgroundShader` uses the zero line (y = 0) to draw the shader which I think causes the issue in my case, because of the `adaptiveYValues` I'm using. I then tried to use the new `TopBottomShader` with a `splitY` value set to the first Y value of my dataset but I'm getting the same result shown above. Though, I really like the new TopBottomShader and what is possible to do with it and I'd like to use it with `AxisValueOverrider.adaptiveYValues(yFraction = 1f)`.

I might be missing something, do you have any recommendation to give me to get the result I'd like?

**Comment by patrickmichalik** [comment]
Hello! This looks like a bug. We’ve identified the issue and will do our best to release a hotfix within a few hours. Thanks for the report!

**Comment by patrickmichalik** [comment, marked as answer]
[Vico 2.0.0 Alpha 3](https://github.com/patrykandpatrick/vico/releases/tag/v2.0.0-alpha.3) fixes this. Both `verticalGradient(arrayOf(...))` and `DynamicShaders.fromBrush(Brush.verticalGradient(listOf(...)))` should produce the desired result. You can also use `TopBottomShader`, which you’ve mentioned, but please note that the split is always at the zero line—`LineCartesianLayer` modifies `splitY`. We’ll be looking into making this customizable. Let me know if you have any questions. Cheers!

**Reply by jibe-n** [reply]
Thank you so much it works like a charm now. You guys rock!

**Reply by patrickmichalik** [reply]
I’m glad to hear that. You’re welcome!
